packages/app/src/components/ErrorModal.tsx

```diff
@@ -1,12 +1,113 @@
-import { Dialog, DialogTitle, DialogContent, DialogContentText } from "@mui/material"
+import { useState } from "react";
+import { Dialog, DialogTitle, DialogContent, DialogContentText, FormControl, InputLabel, Checkbox, ListItemText, MenuItem, OutlinedInput, Select, SelectChangeEvent, Button } from "@mui/material"
+import { useRecoilValue, useSetRecoilState } from "recoil";
+import { dataSelector, encodedDataAtom } from "../atoms/dataAtom";
+import { CoreUpsetData, process } from "@visdesignlab/upset2-core";
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+};
 
 export const ErrorModal = () => {
+    const data = useRecoilValue(dataSelector);
+    const setEncodedDataAtom = useSetRecoilState(encodedDataAtom);
+    const [ encodeList, setEncodeList ] = useState<string[]>([]);
+    let needOneHot = false;
+
+    if (data === null) return null;
+
+    if (Object.values(data.columnTypes).includes('category')) {
+        needOneHot = true;
+    }
+
+    const handleChange = (e: SelectChangeEvent) => {
+        const {
+            target: { value },
+          } = e;
+        setEncodeList(typeof value === 'string' ? value.split(',') : value)
+    }
+
+    const oneHotEncode = (empty?: boolean) => {
+        // close the error window and rerender screen by updating the atom
+        if (empty) {
+            setEncodedDataAtom(null);
+        }
```

Comment:
This seems a bit hacky. Can we close the modal another way? Also, shouldn't the function return here if we do it this way?

Reply:
Newest commit addresses this. The issue was that the `null` value didn't actually change the value and so `Body.tsx` wasn't reactive to the change. By setting this value to the data clone and returning, `Body.tsx` updated in a responsive way.